CMakeLists.txt

```diff
@@ -341,11 +338,25 @@ foreach (lib support kernel search int set float
         set(sources ${sources} ${src})
       endif ()
     endforeach ()
-    add_library(gecode${lib} ${sources} ${${libupper}HDR})
-    target_include_directories(gecode${lib}
-      PUBLIC ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR} )
-    list(APPEND GECODE_INSTALL_TARGETS gecode${lib})
+    add_library(${lib}  ${sources} ${${libupper}HDR})
+    # target_include_directories(${lib}
+    #   PUBLIC ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR} )
+    target_include_directories(${lib}
+    PUBLIC
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+      $<BUILD_INTERFACE: ${CMAKE_CURRENT_SOURCE_DIR}>
+      $<INSTALL_INTERFACE:include>
+  )
+    list(APPEND GECODE_INSTALL_TARGETS ${lib})
   endif ()
+  # Install the binaries
+  install(TARGETS ${lib} EXPORT gecode-targets
+    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib
+    RUNTIME DESTINATION bin
+  )
+  target_compile_features(${lib} PUBLIC cxx_std_17)
+  set_target_properties(${lib} PROPERTIES OUTPUT_NAME gecode_${lib})
```

Comment:
This should be `set_target_properties(${lib} PROPERTIES OUTPUT_NAME gecode${lib})` to make sure that the names of the libraries on the system don't change.

Reply:
done. @Dekker1